src/reducers/SideBarReducer.js

```diff
@@ -1,16 +1,58 @@
-import { Map } from 'immutable';
+import { Map, List } from 'immutable';
 
 const INITIAL_STATE = Map({
-    currentSelectedCity: 'All cities',
+    'sideBarShown': true,
+    'currentSelectedCity': 'All cities',
+    'currentSelectedCategories': List(['All categories']),
+    'hasImages': false,
+    'newest': false
 });
 
 function SideBarReducer (state = INITIAL_STATE, action) {
     switch(action.type){
+    case('TOGGLE_SIDE_BAR'):
+        return state.set('sideBarShown', !state.get('sideBarShown'))
+    case('UPDATE_CATEGORY'):
+        return updateCategory(state, action);
     case('UPDATE_CITY'):
         return state.set('currentSelectedCity', action.data)
+    case('TOGGLE_HAS_IMAGES'):
+        return state.set('hasImages', !state.get('hasImages'))
+    case('TOGGLE_NEWEST'):
+        return state.set('newest', !state.get('newest'))
+    case('RESET_SELECTIONS'):
+        return state.set('currentSelectedCity', 'All cities').set('currentSelectedCategories', List(['All categories']));
     default:
         return state
     }
 }
 
+function updateCategory(state, action) {
+    //if all is selected again, then no changes to the category list
+    if (action.data === 'All categories') {
```

Comment:
Should be a constant.

Any string that is used for logic (e.g., comparison, inclusion, etc) should be a constant. The only string's that don't ahve to be constants are the ones that are shown to the user (e.g., "Click here"). If they are both, they should be a constant.